permission_handler/android/src/main/java/com/baseflow/permissionhandler/PermissionManager.java

```diff
@@ -111,38 +213,37 @@ void requestPermissions(
                     } else {
                         requestResults.put(permission, PermissionConstants.PERMISSION_STATUS_DENIED);
                     }
+                    // On Android below R, the android.permission.MANAGE_EXTERNAL_STORAGE flag in AndroidManifest.xml
+                    // may be ignored and not visible to the App as it's a new permission setting as a whole.
+                    if (permission == PermissionConstants.PERMISSION_GROUP_MANAGE_EXTERNAL_STORAGE && Build.VERSION.SDK_INT < Build.VERSION_CODES.R) {
+                        requestResults.put(permission, PermissionConstants.PERMISSION_STATUS_RESTRICTED);
+                    } else {
+                        requestResults.put(permission, PermissionConstants.PERMISSION_STATUS_DENIED);
+                    }
                 }
 
                 continue;
             }
 
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && permission == PermissionConstants.PERMISSION_GROUP_IGNORE_BATTERY_OPTIMIZATIONS) {
-                activityRegistry.addListener(
-                        new ActivityResultListener(successCallback)
-                );
-
                 String packageName = activity.getPackageName();
                 Intent intent = new Intent();
                 intent.setAction(Settings.ACTION_REQUEST_IGNORE_BATTERY_OPTIMIZATIONS);
                 intent.setData(Uri.parse("package:" + packageName));
                 activity.startActivityForResult(intent, PermissionConstants.PERMISSION_CODE_IGNORE_BATTERY_OPTIMIZATIONS);
+            } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R && permission == PermissionConstants.PERMISSION_GROUP_MANAGE_EXTERNAL_STORAGE) {
+                String packageName = activity.getPackageName();
+                Intent intent = new Intent();
+                intent.setAction(Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION);
+                intent.setData(Uri.parse("package:" + packageName));
+                activity.startActivityForResult(intent, PermissionConstants.PERMISSION_CODE_MANAGE_EXTERNAL_STORAGE);
```

Comment:
It would be good to extract this code into a separate method to avoid repeating code. I am thinking something like:

```java
private void executeIntent(int action, int requestCode) {
    String packageName = activity.getPackageName();
    Intent intent = new Intent();
    intent.setAction(action);
    intent.setData(Uri.parse("package:" + packageName));
    activity.startActivityForResult(intent, requestCode);
}
```

And then we can call it something like this:
```suggestion
                executeIntent(
                    Settings.ACTION_MANAGE_APP_ALL_FILES_ACCESS_PERMISSION, 
                    PermissionConstants.PERMISSION_CODE_MANAGE_EXTERNAL_STORAGE);
```

Reply:
Good call, change has been made